### PR TITLE
Adding file type recognition

### PR DESCRIPTION
**Pull request by VanDavv**

### CODE REVIEW

File: backend/app/src/main/java/com/bls/patronage/cv/CVRequest.java [hunk 1]
```diff
@@ -14,10 +15,10 @@ public CVRequest(URI location, String action) {
         this.action = action;
     }
 
-    public static CVRequest createRecognizeRequest(final URI publicURLToUploadedFile) {
+    public static CVRequest createRecognizeRequest(final URI publicURLToUploadedFile, String givenType) {
```
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/cv/CVRequest.java]: Rename to `fileType`.

File: backend/app/src/main/java/com/bls/patronage/api/AcceptableFileTypes.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package com.bls.patronage.api;
+
+public class AcceptableFileTypes {
+    public static final String IMAGE = "image";
+    public static final String TEXT = "text";
+
+    private final boolean validType;
+    private final String givenType;
```
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/api/AcceptableFileTypes.java]: Rename to `fileType`.

File: backend/app/src/main/java/com/bls/patronage/resources/DecksCvMagicResource.java [hunk 3]
```diff
@@ -43,23 +45,30 @@ public DecksCvMagicResource(StorageService storageService, DeckDAO deckDAO, Flas
     @POST
     @Consumes(MediaType.MULTIPART_FORM_DATA)
     public Response uploadFile(@Auth User user,
-                               @FormDataParam("file") InputStream inputStream) throws StorageException {
+                               @FormDataParam("file") InputStream inputStream,
+                               @QueryParam("fileType") AcceptableFileTypes type) throws StorageException {
+
+        if (!type.isValidType())
```
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/resources/DecksCvMagicResource.java]: Use: `@ValidationMethod` instead (http://www.dropwizard.io/0.9.0/docs/manual/validation.html)

File: backend/app/src/main/java/com/bls/patronage/api/AcceptableFileTypes.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package com.bls.patronage.api;
+
+public class AcceptableFileTypes {
+    public static final String IMAGE = "image";
+    public static final String TEXT = "text";
+
+    private final boolean validType;
```
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/api/AcceptableFileTypes.java]: Remove this flag.

File: backend/app/src/main/java/com/bls/patronage/resources/DecksCvMagicResource.java [hunk 5]
```diff
@@ -43,23 +45,30 @@ public DecksCvMagicResource(StorageService storageService, DeckDAO deckDAO, Flas
     @POST
     @Consumes(MediaType.MULTIPART_FORM_DATA)
     public Response uploadFile(@Auth User user,
-                               @FormDataParam("file") InputStream inputStream) throws StorageException {
+                               @FormDataParam("file") InputStream inputStream,
+                               @QueryParam("fileType") AcceptableFileTypes type) throws StorageException {
```
Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/resources/DecksCvMagicResource.java]: Should that parameter be optional and defaults to image when missing?

Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/resources/DecksCvMagicResource.java]: Just a thought, if this parameter is mandatory, why isn't just an enum?

Comment (VanDavv) [inline backend/app/src/main/java/com/bls/patronage/resources/DecksCvMagicResource.java]: Jersey requirement. It cannot be enum, in this case the best solution was to create class with String constructor

Comment (VanDavv) [inline backend/app/src/main/java/com/bls/patronage/resources/DecksCvMagicResource.java]: Also - I don't think that there should be any default type. We don't know which type will be mostly used

Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/resources/DecksCvMagicResource.java]: > Jersey requirement. It cannot be enum...

What is the exact problem when it's an enum?

Comment (VanDavv) [inline backend/app/src/main/java/com/bls/patronage/resources/DecksCvMagicResource.java]: It's not acceptable. See http://stackoverflow.com/a/11535647/5494277

Comment (mlotysz) [inline backend/app/src/main/java/com/bls/patronage/resources/DecksCvMagicResource.java]: DID YOU TRIED MY PROPOSAL? :) WHAT IS THE RESULT?

Comment (VanDavv) [inline backend/app/src/main/java/com/bls/patronage/resources/DecksCvMagicResource.java]: I'm sorry, it's late and I'm very tired. Enum worked, just figured out that this question was about collection of enum values. New commit is here :)

**Comment by mlotysz** [comment]
Thanks!

**Pull request merged**